### ISSUE TITLE
Migrate private registry e2e tests away from gcr.io/k8s-authenticated-test

### ISSUE THREAD

**Issue by spiffxp** [kubernetes/kubernetes#97026]
Part of umbrella to migrate kubernetes e2e test images/registries to community-owned infrastructure: https://github.com/kubernetes/k8s.io/issues/1458

Followup to https://github.com/kubernetes/kubernetes/issues/97002

The registry is used by the following kubernetes e2e tests:
- [sig-apps] ReplicaSet should serve a basic image on each replica with a private image
- [sig-apps] ReplicationController should serve a basic image on each replica with a private image

https://github.com/kubernetes/k8s.io/issues/1459 covers setting up a community-owned alternative

This issue covers updating the tests.  Fixes may need to be cherry-picked back to previous release branches.

**Comment by spiffxp** [comment]
/sig apps
test owner
/sig testing
/sig release
/area release-eng
/wg k8s-infra

**Comment by spiffxp** [comment]
It's worth reconsidering whether these test have value as currently written

https://github.com/kubernetes/kubernetes/blob/a744bddbf497ab718a40bee1c730e7e848e471ed/test/e2e/apps/rc.go#L68-L73:
- if someone mirror this image into a public registry, they'll still pass
- they don't run unless the [e2e test framework's provider](https://github.com/kubernetes/kubernetes/blob/a744bddbf497ab718a40bee1c730e7e848e471ed/test/e2e/framework/test_context.go#L94-L95) is `gce` or `gke`; practically speaking this is only going to run against clusters created by `cluster/kube-up.sh`

I would be in favor of deleting the tests.  They're testing out that a cluster's nodes have been configured to pull from a known private registry, without the use of ImagePullSecrets.  That's pretty cluster-implementation-specific.

**Comment by spiffxp** [comment]
/sig node
I think node may be a more appropriate test owner for this functionality if we decide we want the test to stick around

**Comment by desponda** [comment]
@spiffxp do you know if this is still targeted to 1.21 Milestone?

**Comment by spiffxp** [comment]
Yes. This is subject to Test Freeze, not Code Freeze. If we find it's unreasonable to land by then, I'll push to next cycle.

**Comment by palnabarun** [comment]
@spiffxp -- friendly heads up, Tomorrow March 24th Wednesday is the v1.21 Test Freeze. :)

**Comment by voigt** [comment]
Due to v1.22 code freeze, I'm moving this issue to the next release.

/milestone v1.23

**Comment by justaugustus** [comment]
@spiffxp -- I haven't been actively tracking this lately... do you know if anyone has done work here recently?
/milestone v1.24

**Comment by spiffxp** [comment]
Last I recall @hh had discussed the idea of setting up an in-cluster private registry with SIG Node (thus making the tests in question more portable) but I don't know if that ever went anywhere

**Comment by jyotimahapatra** [comment]
Hi 👋 Im checking in from bug triage team for release 1.24. Should we track this issue for release 1.24?

**Comment by dims** [comment]
/milestone v1.25

kick the can down the road

**Comment by harshitasao** [comment]
Hi @spiffxp, k8s 1.25 Bug Triage Shadow here, just checking in to see if this is still on track for k8s 1.25?
Thank you.

**Comment by harshitasao** [comment]
Hi, just checking in to see if this is still on track for k8s 1.25?
Thanks!

**Comment by helayoty** [comment]
@dims 👋 the release 1.25 bug triage Lead. I'm reaching out to see the status of this issue and if we are still targeting the current release.

**Comment by dims** [comment]
/milestone v1.26

kick the can down the road yet again!

**Comment by salaxander** [comment]
Hey @dims and @spiffxp - 1.26 bug triage team here! Just wanted to check in and see how we're looking for this release?

Thanks!

**Comment by dims** [comment]
/milestone v1.27

kick the can down the road yet again! :) thanks for the ping @salaxander

**Comment by furkatgofurov7** [comment]
Hey @spiffxp - 1.27 Bug Triage Team Shadow here! 

Just wanted to check if this is still on track for 1.27 release?

Thanks!

> @dims: The provided milestone is not valid for this repository. Milestones in this repository: [`next-candidate`, `v1.16`, `v1.17`, `v1.18`, `v1.19`, `v1.20`, `v1.21`, `v1.22`, `v1.23`, `v1.24`, `v1.25`, `v1.26`]
> 
> Use `/milestone clear` to clear the milestone.

@dims Hi! Looks like the milestone was not properly applied in your previous comment, should we re-apply it? 

Thanks!

**Comment by dims** [comment]
@furkatgofurov7 let's leave it as-is (not apply milestone at all)

**Comment by BenTheElder** [comment]
This either must be done or the tests must be deleted because this is going to start failing when GCR shuts down early next year anyhow (I'm NOT going to migrate it to Artifact Registry in the same google.com account and kick the can further down the road)

**Comment by dims** [comment]
@BenTheElder filed https://github.com/kubernetes/kubernetes/pull/124875 to clean up.

**Comment by tallclair** [comment]
/assign @vivzbansal @tallclair 
Vivek from my team has offered to pick this up (he's not a k8s org member yet, so probably can't actually be assigned).

**Comment by BenTheElder** [comment]
> Note that only [kubernetes members](https://github.com/orgs/kubernetes/people) with read permissions, repo collaborators and people who have commented on this issue/PR can be assigned

[we should rephrase the bot comment, it's a github-side limitation, not a choice on our part, please do work on this!]

**Comment by haircommander** [comment]
I am going to close this to track in https://github.com/kubernetes/kubernetes/issues/130271 since we're more actively working on it there.

/close
